### PR TITLE
Upgrade to Rails 6

### PR DESCRIPTION
**Pull request by cdccollins**
Upgrade app to Rails 6, including switching to Google Chrome in headless mode rather than phantomjs, which does not seem to be maintained at the moment. Also introduced a workaround to avoid an unreported bug in Rails (see commit https://github.com/alphagov/collections/commit/ee70cfc2cf4cbf33fcf03549fa9a3065fdc1e89a). 

:warning: This application is Continuously Deployed: :warning:

- Merged changes are automatically deployed to staging and production.

- Make sure you follow [the guidance for deployments](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) **before** you merge.

- Check your branch is being deployed in the [Release app](https://release.publishing.service.gov.uk/applications/collections), after merging.

### CODE REVIEW

File: config/environments/test.rb [hunk 1]
```diff
@@ -16,26 +16,26 @@
...
 
   # Raise exceptions instead of rendering exception templates.
   config.action_dispatch.show_exceptions = false
 
   # Disable request forgery protection in test environment.
   config.action_controller.allow_forgery_protection = false
 
-  # Raises error for missing translations
-  # config.action_view.raise_on_missing_translations = true
-  config.active_support.test_order = :random
```
Comment (benbaumann95) [inline config/environments/test.rb:34]: Looks as if the PR takes this line out but was there a reason this was added in the first place?

Comment (cdccollins) [reply config/environments/test.rb:34]: That defaults to random in any case, so seems unnecessary?

**Comment by benthorner** [comment]
:warning: This application is Continuously Deployed: :warning:

- Merged changes are automatically deployed to staging and production.

- Make sure you follow [the guidance for deployments](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) **before** you merge.

- Check your branch is being deployed in the [Release app](https://release.publishing.service.gov.uk/applications/collections), after merging.

File: spec/javascripts/support/jasmine.yml [hunk 2]
```diff
@@ -1,34 +1,146 @@
...
+# EXAMPLE
+#
+# rack_options:
+#   server: 'thin'
 
-# path to directory of temporary files
-# (spec runner and asset cache)
-# defaults to tmp/jasmine
-tmp_dir: "tmp/jasmine"
+# phantom_cli_options
```
Comment (Rhian) [inline spec/javascripts/support/jasmine.yml:127]: Given that we've removed the phantomjs dependency, perhaps we could also remove these commented lines?

File: test/test_helper.rb [hunk 3]
```diff
@@ -15,7 +15,7 @@
 require "mocha/minitest"
 
 require "webmock/minitest"
-WebMock.disable_net_connect!
+WebMock.disable_net_connect!(allow_localhost: true)
```
Comment (Rhian) [inline test/test_helper.rb:18]: I notice in this commit `https://github.com/alphagov/travel-advice-publisher/pull/919/commits/bd74aa591b5e085d1436c51b0ef69773f3f19579` that the webdrivers gem is explicitly set to allow new Chromedrivers to be downloaded. Perhaps we could do something similar?

**Pull request merged**